packages/server/lib/browsers/firefox.ts

```diff
@@ -458,7 +458,10 @@ export async function open (browser: Browser, url: string, options: BrowserLaunc
   const [
     foxdriverPort,
     marionettePort,
-  ] = await Promise.all([getPort(), getPort()])
+  ] = await Promise.all([
+    Number(process.env.CYPRESS_FOXDRIVER_PORT) || getPort(),
+    Number(process.env.CYPRESS_MARIONETTE_PORT) || getPort(),
```

Comment:
Allow Dynamic Firefox Port:

> >The ideal solution would be to allow the browser to bind to a random port and we pull this dynamic port from the browser. You are right that we can do this for the marionette port. I was able to draft this up on this branch: [emily (compare)](https://github.com/cypress-io/cypress/compare/emily/ff-dynamic-marionette-port)

> Thank you that looks good and seems to work for me as well.

Let's move forward with the draft changes I wrote up for allowing firefox to handle the dynamic port allocation.  Sounds like this works for you 👍🏻